client/src/app/components/FilterToolbar/SelectFilterControl.tsx

```diff
@@ -91,15 +75,25 @@ export const SelectFilterControl = <TItem, TFilterCategoryKey extends string>({
       <Select
         className={css(isScrollable && "isScrollable")}
         aria-label={category.title}
-        toggleId={`${category.key}-filter-value-select`}
-        onToggle={() => setIsFilterDropdownOpen(!isFilterDropdownOpen)}
-        selections={selections || []}
-        onSelect={(_, value) => onFilterSelect(value)}
+        toggle={toggle}
+        onOpenChange={(isOpen) => setIsFilterDropdownOpen(isOpen)}
+        selected={filterValue}
+        onSelect={(_, value) => {
+          onFilterSelect(value as string);
+          setIsFilterDropdownOpen(false);
+        }}
         isOpen={isFilterDropdownOpen}
-        placeholderText="Any"
-        isDisabled={isDisabled || category.selectOptions.length === 0}
+        placeholder="Any"
       >
-        {renderSelectOptions(category.selectOptions)}
+        <SelectList>
+          {category.selectOptions.map((o: SelectOptionProps, index) => {
+            return (
+              <SelectOption {...o} key={`${index}-${o.value}`}>
+                {o.value}
+              </SelectOption>
```

Comment:
I took a look at the filters across the codebase that uses the single select to see what the data coming in look like.  Consider:

https://github.com/konveyor/tackle2-ui/blob/6217116e9c8c7d0481d12ae2b0f7e83940f88724/client/src/app/pages/review/components/application-assessment-summary-table/application-assessment-summary-table.tsx#L77-L95

Then the code there probably won't work since the `selectOptions` key is ignored.

Maybe more like this:
```tsx
  <SelectOption
    {...o}
    key={`${index}-${o.value}`}
    value={o.key}
  >
    {o.value}
  </SelectOption>
```

and then the `chips` would also probably need to be adjusted to convert the selected "key"(aka SelectOption value) to the display "value" (aka SelectOption children).

Reply:
Updated to address this.